client/my-sites/upgrades/domain-management/transfer/transfer-to-other-user/index.jsx

```diff
@@ -0,0 +1,218 @@
+				closeDialog();
+			}, err => {
+				this.setState( { disableDialogButtons: false } );
+				this.props.errorNotice( err.message || defaultErrorMessage );
+				closeDialog();
+			} );
+	}
+
+	handleDialogClose() {
+		if ( ! this.state.disableDialogButtons ) {
+			this.setState( { showConfirmationDialog: false } );
+		}
+	}
+
+	getSelectedUserDisplayName() {
+		const selectedUser = find( this.props.users, user => user.ID === Number( this.state.selectedUserId ) );
+		if ( ! selectedUser ) {
+			return '';
+		}
+		return this.getUserDisplayName( selectedUser );
+	}
+
+	getUserDisplayName( { first_name, last_name, nice_name } ) {
+		return ( first_name && last_name ) ? `${ first_name } ${ last_name } (${ nice_name })` : nice_name;
+	}
+
+	render() {
+		if ( ! this.isDataReady() ) {
+			return <DomainMainPlaceholder goBack={ this.goToEdit } />;
+		}
+
+		const { selectedSite, selectedDomainName } = this.props,
+			{ slug } = selectedSite;
+
+		return (
+			<Main className="transfer-to-other-user">
+				<Header
+					selectedDomainName={ selectedDomainName }
+					backHref={ paths.domainManagementEdit( slug, selectedDomainName ) }>
+					{ this.props.translate( 'Transfer Domain To Another User' ) }
+				</Header>
+				{ this.renderSection() }
+			</Main>
+		);
+	}
+
+	renderDialog() {
+		const buttons = [
+			{
+				action: 'cancel',
+				label: this.props.translate( 'Cancel' ),
+				disabled: this.state.disableDialogButtons
+			},
+			{
+				action: 'confirm',
+				label: this.props.translate( 'Confirm Transfer' ),
+				onClick: this.handleConfirmTransferDomain,
+				disabled: this.state.disableDialogButtons,
+				isPrimary: true
+			}
+			],
+			domainName = this.props.selectedDomainName,
+			selectedUserDisplay = this.getSelectedUserDisplayName();
+		return (
+			<Dialog isVisible={ this.state.showConfirmationDialog } buttons={ buttons } onClose={ this.handleDialogClose }>
+				<h1>{ this.props.translate( 'Confirm Transfer' ) }</h1>
+				<p>{ this.props.translate( 'Do you want to transfer the ownership of {{strong}}%(domainName)s{{/strong}} ' +
+					'to {{strong}}%(selectedUserDisplay)s{{/strong}}?', {
+						args: { domainName, selectedUserDisplay }, components: { strong: <strong /> }
+					} ) }</p>
+			</Dialog>
+		);
+	}
+
+	renderSection() {
+		const { selectedDomainName: domainName, translate, users } = this.props,
+			availableUsers = this.filterAvailableUsers( users ),
+			{ currentUserCanManage } = getSelectedDomain( this.props );
+
+		if ( ! currentUserCanManage ) {
+			return <NonOwnerCard { ...omit( this.props, [ 'children' ] ) } />;
+		}
+
+		return (
+			<div>
+				<SectionHeader label={ translate( 'Transfer Domain To Another User' ) } />
+				<Card className="transfer-card">
+					<p>
+						{ translate( 'Transferring a domain to another user will give all the rights of the domain to that user. ' +
+							'Please choose a user to the transfer {{strong}}%(domainName)s{{/strong}}.',
```

Comment:
"Please choose a user to the transfer X" doesn't sound right :) "Please choose a user to transfer X to"?
Or even better, use "administrator", instead of "user", to hint of the needed privileges.